tools/package_gcsfuse_docker/Dockerfile

```diff
@@ -31,42 +32,47 @@ ENV GCSFUSE_PATH "$GOPATH/src/$GCSFUSE_REPO"
 RUN go get -d ${GCSFUSE_REPO}
 
 WORKDIR ${GCSFUSE_PATH}
+ARG DEBEMAIL="gcs-fuse-dev@google.com"
+ARG DEBFULLNAME="GCSFuse Team"
+
 # Build Arg for building through a particular branch/commit. By default, it uses
 # the tag corresponding to passed GCSFUSE VERSION
-ARG BRANCH_NAME="v${GCSFUSE_VERSION}"
+#ARG BRANCH_NAME="v${GCSFUSE_VERSION}"
 RUN git checkout "${BRANCH_NAME}"
 
 # Install fpm package using bundle
 RUN bundle install --gemfile=${GCSFUSE_PATH}/tools/gem_dependency/Gemfile
 
-ARG GCSFUSE_BIN="/gcsfuse"
+ARG GCSFUSE_BIN="/gcsfuse_${GCSFUSE_VERSION}_amd64"
+ARG GCSFUSE_DOC="${GCSFUSE_BIN}/usr/share/doc/gcsfuse"
 WORKDIR ${GOPATH}
 RUN go install ${GCSFUSE_REPO}/tools/build_gcsfuse
 RUN mkdir -p ${GCSFUSE_BIN}
 RUN build_gcsfuse ${GCSFUSE_PATH} ${GCSFUSE_BIN} ${GCSFUSE_VERSION}
 RUN mkdir -p ${GCSFUSE_BIN}/usr && mv ${GCSFUSE_BIN}/bin ${GCSFUSE_BIN}/usr/bin
+RUN mkdir -p ${GCSFUSE_BIN}/DEBIAN && cp $GOPATH/src/$GCSFUSE_REPO/DEBIAN/* ${GCSFUSE_BIN}/DEBIAN/
+RUN mkdir -p ${GCSFUSE_DOC}
+RUN mv ${GCSFUSE_BIN}/DEBIAN/copyright ${GCSFUSE_DOC} &&  \
+    mv ${GCSFUSE_BIN}/DEBIAN/changelog ${GCSFUSE_DOC} && \
+    mv ${GCSFUSE_BIN}/DEBIAN/gcsfuse-docs.docs ${GCSFUSE_DOC}
+RUN sed -i "1s/.*/gcsfuse (${GCSFUSE_VERSION}) stable; urgency=medium/" ${GCSFUSE_DOC}/changelog && \
+    sed -i "1s/.*/Version: ${GCSFUSE_VERSION}/" ${GCSFUSE_BIN}/DEBIAN/control
+RUN gzip -9 -n ${GCSFUSE_DOC}/changelog
+RUN strip --strip-unneeded ${GCSFUSE_BIN}/usr/bin/gcsfuse && \
+    strip --strip-unneeded ${GCSFUSE_BIN}/sbin/mount.gcsfuse
+RUN dpkg-deb --build ${GCSFUSE_BIN}
 
 ARG GCSFUSE_PKG="/packages"
 RUN mkdir -p ${GCSFUSE_PKG}
 WORKDIR ${GCSFUSE_PKG}
-RUN fpm \
-    -s dir \
-    -t deb \
-    -n gcsfuse \
-    -C ${GCSFUSE_BIN} \
-    -v ${GCSFUSE_VERSION} \
-    -d fuse \
-    --vendor "" \
-    --url "https://$GCSFUSE_REPO" \
-    --description "A user-space file system for Google Cloud Storage."
+RUN mv ${GCSFUSE_BIN}.deb .
 RUN fpm \
```

Comment:
so license file is added only to apt packages and not rpm packages?

Reply:
Added license flag to rpm package
```
ashmeen@ashmeen:~/gcsfuse/release/v1.0.2/packages$ rpm -qid gcsfuse-1.0.2-1.x86_64.rpm
Name        : gcsfuse
Version     : 1.0.2
Release     : 1
Architecture: x86_64
Install Date: (not installed)
Group       : default
Size        : 22491743
License     : Apache-2.0
Signature   : (none)
Source RPM  : gcsfuse-1.0.2-1.src.rpm
Build Date  : Mon 17 Jul 2023 06:27:21 AM UTC
Build Host  : 47c73e7d027e
Relocations : / 
Packager    : GCSFuse Team <gcs-fuse-dev@google.com>
URL         : https://github.com/googlecloudplatform/gcsfuse/
Summary     : A user-space file system for Google Cloud Storage.
Description :
A user-space file system for Google Cloud Storage.
/usr/share/doc/gcsfuse/changelog.gz
/usr/share/doc/gcsfuse/copyright
/usr/share/doc/gcsfuse/gcsfuse-docs.docs
```
Also verified that installation of rpm package adds copyright file at /usr/share/doc/gcsfuse
```
[starterscriptuser@release-test-rhel-8 gcsfuse]$ ls
changelog.gz  copyright  gcsfuse-docs.docs
```

Reply:
Just adding the license flag generated the copyright files similar to deb?

Reply:
No, this is because of the changed source repository structure. The license flag just adds the license in the metadata as viewed by `rpm -qid gcsfuse-1.0.2-1.x86_64.rpm` command.